src/modules/user/user.usecase.ts

```diff
@@ -27,4 +68,217 @@ export class UserUseCases {
+          password: userPass,
+          mnemonic: newUser.mnemonic,
+          hKey: userSalt,
+          referrer: newUser.referrer,
+          referralCode: v4(),
+          uuid: null,
+          credit: 0,
+          welcomePack: true,
+          // ?
+          // registerCompleted: newUser.registerCompleted,
+          username: newUser.email,
+          bridgeUser: newUser.email,
+        },
+        transaction,
+      });
+
+      if (!isNewRecord) {
+        throw new UserAlreadyRegisteredError(newUser.email);
+      }
+
+      const { userId, uuid: userUuid } = await this.networkService.createUser(
+        email,
+      );
+
+      await this.userRepository.updateById(
+        userResult.id,
+        { userId, uuid: userUuid },
+        transaction,
+      );
+
+      if (hasReferrer) {
+        const event = new InvitationAcceptedEvent(
+          'invitation.accepted',
+          userUuid,
+          referrer.uuid,
+          referrer.email,
+          {},
+        );
+
+        this.notificationService.add(event);
+        await this.sharedWorkspaceRepository.updateByHostAndGuest(
+          referrer.id,
+          email,
+        );
+
+        await this.applyReferral(referrer.id, 'invite-friends');
+      }
+
+      const token = Sign(newUser.email, this.configService.get('secrets.jwt'));
+      const bucket = await this.networkService.createBucket(email, userId);
+      const rootFolderName = await this.cryptoService.encryptName(
+        `${bucket.name}`,
+      );
+      const rootFolder = await this.folderUseCases.createRootFolder(
+        userResult,
+        rootFolderName,
+        bucket.id,
+      );
+
+      await transaction.commit();
+      await this.createUserReferrals(userResult.id);
+      await this.userRepository.updateById(userResult.id, {
+        rootFolderId: rootFolder.id,
+      });
+      await Promise.all([
+        this.folderUseCases.createFolder(userResult, 'Family', rootFolder.id),
+        this.folderUseCases.createFolder(userResult, 'Personal', rootFolder.id),
+      ]);
+
+      return {
+        token,
+        user: {
+          ...userResult.toJSON(),
+          hKey: userResult.hKey.toString(),
+          password: userResult.password.toString(),
+          mnemonic: userResult.mnemonic.toString(),
+          rootFolderId: rootFolder.id,
+          bucket: bucket.id,
+          uuid: userUuid,
+          userId,
+        },
+        uuid: userUuid,
+      };
+    } catch (err) {
+      await transaction.rollback();
+
+      throw err;
+    }
+  }
+
+  async createUserReferrals(userId: UserAttributes['id']): Promise<void> {
+    const referrals = await this.referralsRepository.findAll({ enabled: true });
+    const userReferralsToCreate: Omit<
+      UserReferralAttributes,
+      'id' | 'startDate'
+    >[] = [];
+
+    referrals.forEach((referral) => {
+      Array(referral.steps)
+        .fill(null)
+        .forEach(() => {
+          const applied = referral.key === 'create-account';
```

Comment:
We should centralize the hardcoded values at least in an enum 'create-account', 'complete-survey' and so on

Reply:
Done